util/dvsim/MsgBuckets.py

```diff
@@ -0,0 +1,218 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""
+Helper class for aggregating message categories from multiple tool runs
+"""
+import hjson
+import copy
+import logging as log
+from pathlib import Path
+
+from typing import Dict, List
+from utils import print_msg_list
+
+# Known severity levels and color map for Dvsim summary tables.
+SEV_COLMAP = {
+    'info': 'I',
+    'warning': 'W',
+    'error': 'E'
+}
+
+
+class MsgBuckets():
+    def __init__(self,
+                 bucket_dict: Dict = None,
+                 result_path: Path = None,
+                 report_sev: List[str] = ['warning', 'error'],
+                 fail_sev: List[str] = ['warning', 'error']) -> None:
+        self.buckets = {}
+        self.labels = {}
+        self.severities = {}
+        self.set_sev(report_sev, fail_sev)
+        if bucket_dict is not None:
+            self.load_dict(bucket_dict)
+        elif result_path is not None:
+            self.load_hjson(result_path)
+
+    @staticmethod
+    def validate_dict(bucket_dict: Dict) -> None:
+        """Validates that the required dict keys are present"""
+        check_copy = copy.deepcopy(bucket_dict)
+        for key, val in check_copy.items():
+            if 'label' not in val:
+                raise RuntimeError('Missing label in bucket {}'.format(key))
+            if 'severity' not in val:
+                raise RuntimeError('Missing severity in bucket {}'.format(key))
+            if 'messages' not in val:
+                raise RuntimeError('Missing messages in bucket {}'.format(key))
+            if not isinstance(val['label'], str):
+                raise RuntimeError('Label {} in bucket {} is not a string'
+                                   .format(val['label'], key))
+            if val['severity'] not in SEV_COLMAP:
+                raise RuntimeError('Invalid severity {} in bucket {}'
+                                   .format(val['severity'], key))
+            if not isinstance(val['messages'], List):
+                raise RuntimeError('Entries in bucket {} must be a list '
+                                   'of strings'.format(key))
+            del val['label']
+            del val['severity']
+            del val['messages']
+            if val:
+                raise RuntimeError('Unused keys present in bucket {}'
+                                   .format(key))
+        del check_copy
+
+    def load_hjson(self, result_path: Path) -> None:
+        """Load message buckets from Hjson and append to this instance"""
```

Comment:
Is that hjson from a previous run?

Reply:
I think the comment here is out of date. 
This calls `load_dict` underneath, which clears the internal data structure and initializes with the values in the dict. Will fix.